### ISSUE TITLE
Fdisk warning about GPT partiion 

### ISSUE THREAD

**Issue by erranfenech** [rackerlabs/recap#86]
Hi guys

Wanted to find out if you could assist?

Recap has been configured to notify the recipient  via email of any errors, one of the errors that is constantly been highlighted is that fdisk sees a GPT partition, every time this runs its fires off an email.

Do you have a fix for this? EG:

----------
# fdisk -l /dev/sdb 

WARNING: GPT (GUID Partition Table) detected on '/dev/sdb'! The util fdisk doesn't support GPT. Use GNU Parted.

Disk /dev/sdb: 91.3 GB, 91268055040 bytes
255 heads, 63 sectors/track, 11096 cylinders
Units = cylinders of 16065 * 512 = 8225280 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disk identifier: 0x00000000

   Device Boot      Start         End      Blocks   Id  System
/dev/sdb1               1       11097    89128959+  ee  GPT
----------

# recap 2>&1 >/dev/null

WARNING: GPT (GUID Partition Table) detected on '/dev/sdb'! The util fdisk doesn't support GPT. Use GNU Parted.

-----------

As this is not really an error it would be good if this could be squashed so no email is triggered for this kind of Warning.

Thanks
Erran

**Comment by tonyskapunk** [comment]
Hi @erranfenech, Thanks for reporting this.

Is definitely not an error, in your example `fdisk` is not the right tool for gathering the details of the GPT table for your block device `/dev/sdb`, a quick fix is using `parted` instead as both have similar functionality with `-l` but the output is a bit different.

In this test I have a centos 7 for testing, see how the ```WARNING: fdisk GPT support is currently new, and therefore in an experimental phase. Use at your own discretion.``` for `fdisk` is different, seems like yours is running an older version.

https://gist.github.com/tonyskapunk/30c5c9aad279d9718ffe74230c73ef3a

Super quick fix would be something like this in this [line](https://github.com/rackerlabs/recap/blob/master/src/recap#L385)
```bash
FDISK=$( which parted fdisk 2>/dev/null | head -1 )
${FDISK} ${OPTS_FDISK} >> "${LOGFILE}"
```

The problem lies if the server does not have installed parted, but fdisk would be used as a safety net in that case.

I'll create a PR for this.

**Comment by tonyskapunk** [comment]
I updated https://gist.github.com/tonyskapunk/30c5c9aad279d9718ffe74230c73ef3a with the outputs of `fdisk` and `parted` for:
- centos 7,6
- debian 8,7
- ubuntu 14.04,16.04

From my tests I found that centos 6, debian 7, ubuntu 14.04 and centos 7 sent output to **stderr** when using `fdisk` (see below), most of the outputs suggest using `GNU parted`, then this supports that the preferred option should be `parted` then failback to use `fdisk`.

- el6
```bash
# fdisk -l 1>/dev/null 

WARNING: GPT (GUID Partition Table) detected on '/dev/xvdb'! The util fdisk doesn't support GPT. Use GNU Parted.
```

- debian 7 (wheezy)
```bash
# fdisk -l 1>/dev/null 

WARNING: GPT (GUID Partition Table) detected on '/dev/xvdb'! The util fdisk doesn't support GPT. Use GNU Parted.

Disk /dev/mapper/vg0-test0 doesn't contain a valid partition table
Disk /dev/mapper/vg1-test1 doesn't contain a valid partition table
```

- ubuntu 14.04 (trusty)
```bash
# fdisk -l 1>/dev/null 

WARNING: GPT (GUID Partition Table) detected on '/dev/xvdb'! The util fdisk doesn't support GPT. Use GNU Parted.

Disk /dev/mapper/vg1-test1 doesn't contain a valid partition table
Disk /dev/mapper/vg0-test0 doesn't contain a valid partition table
```

- centos 7
```bash
# fdisk -l 1>/dev/null
WARNING: fdisk GPT support is currently new, and therefore in an experimental phase. Use at your own discretion.
```

**Comment by tonyskapunk** [comment]
Created commit 9dd06a8 on #78 to address this.

An example on a centos 7 looks like this:

### Using fdisk
```bash
# cat /var/log/recap/fdisk_20170428-212625.log 
2017-04-28_21:26:25
Disk partitions

Disk /dev/xvda: 21.5 GB, 21474836480 bytes, 41943040 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disk label type: dos
Disk identifier: 0x000b29af

    Device Boot      Start         End      Blocks   Id  System
/dev/xvda1   *        2048    41943039    20970496   83  Linux
```

### Using parted
```bash
# cat /var/log/recap/fdisk_20170428-213022.log 
2017-04-28_21:30:22
Disk partitions
Model: Xen Virtual Block Device (xvd)
Disk /dev/xvda: 21.5GB
Sector size (logical/physical): 512B/512B
Partition Table: msdos
Disk Flags: 

Number  Start   End     Size    Type     File system  Flags
 1      1049kB  21.5GB  21.5GB  primary  ext4         boot

```

**Comment by tonyskapunk** [comment]
I removed the last commit that "fixed" this, due to potential issues pointed out by @StafDeHat in #78 (https://github.com/rackerlabs/recap/commit/9dd06a80d972dfb2aec32f7345972ad70a2b6246) this will need to be re-evaluated.

**Comment by tonyskapunk** [comment]
Added new commit 02e12c6 , this one simply includes `stderr` into `stdout` to prevent errors, but still logs the Warning message about GPT into the logs. Fix #86
